src/main/java/org/breedinginsight/daos/TraitDAO.java

```diff
@@ -453,6 +454,12 @@ public Trait updateTraitBrAPI(Trait trait, Program program) {
         return updatedTrait;
     }
 
+    private String constructMethodName(Trait trait, Program program) {
+        return !StringUtils.isBlank(trait.getMethod().getDescription()) ?
```

Comment:
Up to you: from a reusability perspective, the `Utilities.appendProgramKey` method could be used to construct the method name